### PR TITLE
Address code review comments on initial commit

### PR DESCRIPTION
**Pull request by bswartz**
Primarily this change removed generated code for the type clients, listers,
and informers, and replaces those functions with dynamic versions to more
closely match what kube builder does. The main benefit was to remove a
dependency on k8s.io/code-generator, and to simplify keeping up to date
with the latest k8s release.

```release-note
Removed generated clients. Please use the dynamic client when writing controllers to interact with VolumePopulator objects.
```

### CODE REVIEW

File: client/config/crd/populator.storage.k8s.io_volumepopulators.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.2.4
-    api-approved.kubernetes.io: unapproved
+    controller-gen.kubebuilder.io/version: v0.5.0
+    api-approved.kubernetes.io: https://github.com/kubernetes-csi/volume-data-source-validator/pull/0
```
Comment (xing-yang) [inline client/config/crd/populator.storage.k8s.io_volumepopulators.yaml:8]: pull 0 doesn't exist?  Probably better to leave it blank and leave a comment that it was committed without a pull number.  Maybe you can add the commit id in the comment for reference.

File: go.mod [hunk 2]
```diff
@@ -3,36 +3,25 @@ module github.com/kubernetes-csi/volume-data-source-validator
...
 
 replace (
 	github.com/kubernetes-csi/volume-data-source-validator/client => ./client
-	k8s.io/api => k8s.io/api v0.19.9
-	k8s.io/apimachinery => k8s.io/apimachinery v0.19.9
-	k8s.io/client-go => k8s.io/client-go v0.19.9
-	k8s.io/code-generator => k8s.io/code-generator v0.19.9
+	k8s.io/api => k8s.io/api v0.21.0
+	k8s.io/apimachinery => k8s.io/apimachinery v0.21.0
+	k8s.io/client-go => k8s.io/client-go v0.21.0
```
Comment (xing-yang) [inline go.mod:26]: Can you see if these k8s.io replacements can be removed?  I think they can be removed if go.mod does not need k8s.io/kubernetes.

File: client/config/crd/populator.storage.k8s.io_volumepopulators.yaml (hunk 1 above)
Comment (bswartz) [reply client/config/crd/populator.storage.k8s.io_volumepopulators.yaml:8]: It has to be the word "unapproved" or a URL pointing to a document (enforced by API server). I suppose I can point to the KEP document.

File: go.mod (hunk 2 above)
Comment (bswartz) [reply go.mod:26]: You're right.

**Comment by xing-yang** [comment]
Can you squash the commits?

File: client/config/crd/populator.storage.k8s.io_volumepopulators.yaml (hunk 1 above)
Comment (xing-yang) [reply client/config/crd/populator.storage.k8s.io_volumepopulators.yaml]: Sure.

**Comment by xing-yang** [comment]
We can get this PR merged once you've squashed the commits.

**Comment by bswartz** [comment]
Squashed!

**Comment by bswartz** [comment]
Err wait, the squash failed

**Comment by bswartz** [comment]
Okay I double checked. The squash worked this time.

**Pull request merged**